### PR TITLE
Added basic prototype funcionality

### PR DESCRIPTION
**Pull request by HoundThe**

### CODE REVIEW

File: WebApplication1.sln [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApplication1", "WebApplication1\WebApplication1.csproj", "{D69A989B-5733-4E1D-940D-7E498293CA6B}"
```
Comment (RheaAyase) [inline WebApplication1.sln]: Let's rename the project&solution to `VerificationWeb` maybe?

File: WebApplication1/Services/RoleService.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            _guild = client.Guilds.FirstOrDefault(x => x.Id == Config.GuildId);
+        }
+
+        public async Task AssignRoleAsync(ulong userId, IEnumerable<string> roles)
+        {
+            // TODO find the nullref bug
+            try
+            {
+                var rolesToAdd = new List<IRole>();
+                foreach (var x in roles)
```
Comment (RheaAyase) [inline WebApplication1/Services/RoleService.cs]: Try to not (ab)use `var` too much. If I were to take this one line out of context (for example when looking at diff) I wouldn't know what type is `x` here.

File: WebApplication1/Services/RoleService.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        public RoleService(Config config, DiscordSocketClient client)
+        {
+            Config = config;
+            _guild = client.Guilds.FirstOrDefault(x => x.Id == Config.GuildId);
+        }
+
+        public async Task AssignRoleAsync(ulong userId, IEnumerable<string> roles)
+        {
+            // TODO find the nullref bug
```
Comment (RheaAyase) [inline WebApplication1/Services/RoleService.cs]: Found it?

File: WebApplication1/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    x.ClientId = Config.RedditId;
+                    x.ClientSecret = Config.RedditSecret;
+                    x.ClaimActions.MapJsonKey("id", "id");
+                    x.ClaimActions.MapJsonKey("name", "name");
+                    x.CallbackPath = "/signin-reddit";
+                    x.Scope.Add("identity");
+                });
+            // TODO remove hardcoded URI
+            var webAgent = new BotWebAgent(Config.RedditBotName, Config.RedditBotPassword, Config.RedditBotId,
+                Config.RedditBotSecret, "https://127.0.0.1:5001/");
```
Comment (RheaAyase) [inline WebApplication1/Startup.cs]: TODO :P

**Review by RheaAyase — CHANGES_REQUESTED** [review]

File: WebApplication1.sln (hunk 1 above)
Comment (HoundThe) [reply WebApplication1.sln]: Renamed

File: WebApplication1/Services/RoleService.cs (hunk 3 above)
Comment (HoundThe) [reply WebApplication1/Services/RoleService.cs]: I think so, couldn't replicate it now, removed the comment

File: WebApplication1/Services/RoleService.cs (hunk 2 above)
Comment (HoundThe) [reply WebApplication1/Services/RoleService.cs]: Noted, I am being tempted by R# screaming to change everything to var :D Changed var to string and x to a proper name

File: WebApplication1/Startup.cs (hunk 4 above)
Comment (HoundThe) [reply WebApplication1/Startup.cs]: Done

**Pull request merged**